autogen/agentchat/contrib/huggingface_utils.py

```diff
@@ -0,0 +1,186 @@
+import os
+from typing import TYPE_CHECKING, Any, Dict, Literal, Optional, Union
+
+import torch
+from huggingface_hub import ImageToTextOutput, InferenceClient
+from PIL.Image import Image
+
+from autogen.agentchat.contrib import img_utils
+
+
+class HuggingFaceClient:
```

Comment:
Is this meant to be a model client?
https://github.com/microsoft/autogen/blob/19de99e3f6e46f6040d54c4a55785d02158dec28/autogen/oai/client.py#L64

Reply:
It did not implement the model client protocol yet. But it's a good suggestion. I'll make the change.

Reply:
I have implemented the `HuggingFaceClient` following `ModelClient` protocol, could you help review again?

Some notes:
1. Currently the client only supports the vision related tasks. My next step is to add support for text completion, so that we can use it like `OpenAIClient` and `GeminiClient`. (Maybe in a separate PR, as the current PR is already too large to review efficiently)
2. We need to pass an additional `task` parameter to specify which task shall be called in `HuggingFaceClient.create()` method. Do you have any better suggestion to support client multi-task behavior?
3. I haven't add UT yet.
4. The cost of HuggingFace call is hard to be determined. Therefore I left it with 0.0 as a default value.